Bug：compile error with native build

**Summary**

- how to reproduce:


OS: Linux
cpu: with sse4_2 capability

`make build-native`

~~~
Compiling common-io v0.1.0 (/home/wu/databend/common/io)
   Compiling common-cache v0.1.0 (/home/wu/databend/common/cache)
error[E0308]: mismatched types
   --> common/io/src/[position.rs:384](http://position.rs:384/):36
    |
384 |                     return index + _mm_cmpestri::<0>(chars_set, chars_count, bytes, 16);
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `i32`

error[E0277]: cannot add `i32` to `usize`
   --> common/io/src/[position.rs:384](http://position.rs:384/):34
    |
384 |                     return index + _mm_cmpestri::<0>(chars_set, chars_count, bytes, 16);
    |                                  ^ no implementation for `usize + i32`
    |
    = help: the trait `Add<i32>` is not implemented for `usize`
    = help: the following other types implement trait `Add<Rhs>`:
              <&'a f32 as Add<f32>>
              <&'a f64 as Add<f64>>
              <&'a i128 as Add<i128>>
              <&'a i128 as Add<num_bigint::bigint::BigInt>>
              <&'a i16 as Add<i16>>
              <&'a i16 as Add<num_bigint::bigint::BigInt>>
              <&'a i32 as Add<i32>>
              <&'a i32 as Add<num_bigint::bigint::BigInt>>
            and 192 others

error[E0308]: mismatched types
   --> common/io/src/[position.rs:389](http://position.rs:389/):27
    |
389 |                           + _mm_cmpestri::<_SIDD_NEGATIVE_POLARITY>(
    |  ___________________________^
390 | |                             chars_set,
391 | |                             chars_count,
392 | |                             bytes,
393 | |                             16,
394 | |                         );
    | |_________________________^ expected `usize`, found `i32`

error[E0277]: cannot add `i32` to `usize`
   --> common/io/src/[position.rs:389](http://position.rs:389/):25
    |
389 |                         + _mm_cmpestri::<_SIDD_NEGATIVE_POLARITY>(
    |                         ^ no implementation for `usize + i32`
    |
    = help: the trait `Add<i32>` is not implemented for `usize`
    = help: the following other types implement trait `Add<Rhs>`:
              <&'a f32 as Add<f32>>
              <&'a f64 as Add<f64>>
              <&'a i128 as Add<i128>>
              <&'a i128 as Add<num_bigint::bigint::BigInt>>
              <&'a i16 as Add<i16>>
              <&'a i16 as Add<num_bigint::bigint::BigInt>>
              <&'a i32 as Add<i32>>
              <&'a i32 as Add<num_bigint::bigint::BigInt>>
            and 192 others

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `common-io` due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
All done...
~~~


- CPU Info

~~~
~/databend$ lscpu | grep sse
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 
sse4_1 sse4_2 
movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
~~~

--- a/common/io/src/position.rs
+++ b/common/io/src/position.rs
@@ -381,17 +381,13 @@ fn position_sse42<
 
             if POSITIVE {
                 if _mm_cmpestrc::<0>(chars_set, chars_count, bytes, 16) > 0 {
-                    return index + _mm_cmpestri::<0>(chars_set, chars_count, bytes, 16);
+                    return index + _mm_cmpestri::<0>(chars_set, chars_count, bytes, 16) as usize;
                 }
             } else {
                 if _mm_cmpestrc::<_SIDD_NEGATIVE_POLARITY>(chars_set, chars_count, bytes, 16) > 0 {
                     return index
-                        + _mm_cmpestri::<_SIDD_NEGATIVE_POLARITY>(
-                            chars_set,
-                            chars_count,
-                            bytes,
-                            16,
-                        );
+                        + _mm_cmpestri::<_SIDD_NEGATIVE_POLARITY>(chars_set, chars_count, bytes, 16)
+                            as usize;
                 }
             }
 

